packages/metro-service/src/asset/ios.ts

```diff
@@ -68,17 +70,48 @@ export function writeImageSet(imageSet: ImageSet): void {
   );
 }
 
-export function getAssetDestPathIOS(
-  asset: PackagerAsset,
-  scale: number
-): string {
-  const suffix = scale === 1 ? "" : `@${scale}x`;
-  const fileName = `${asset.name + suffix}.${asset.type}`;
-  return path.join(
-    // Assets can have relative paths outside of the project root.
-    // Replace `../` with `_` to make sure they don't end up outside of
-    // the expected assets directory.
-    asset.httpServerLocation.substring(1).replace(/\.\.\//g, "_"),
-    fileName
-  );
+const ALLOWED_SCALES = [1, 2, 3];
+
+export function saveAssetsIOS(
+  assets: ReadonlyArray<AssetData>,
+  _platform: string,
+  _assetsDest: string | undefined,
+  assetCatalogDest: string | undefined,
+  addAssetToCopy: (
+    asset: AssetData,
+    allowedScales: number[] | undefined,
+    getAssetDestPath: (asset: AssetData, scale: number) => string
+  ) => void
+) {
```

Comment:
Nit: If we make this an arrow function, we can type it properly:
```suggestion
export const saveAssetsIOS: SaveAssetsPlugin = (
  assets,
  _platform,
  _assetsDest,
  addAssetToCopy
) => {
```